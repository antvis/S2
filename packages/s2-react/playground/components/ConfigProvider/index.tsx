import { getLang, type ThemeName } from '@antv/s2';
import { ConfigProvider as AntdConfigProvider, theme } from 'antd';
import enUS from 'antd/es/locale/en_US';
import ruRU from 'antd/es/locale/ru_RU';
import zhCN from 'antd/es/locale/zh_CN';
import React from 'react';

export interface ConfigProviderProps {
  children: React.ReactNode;
  themeName?: ThemeName;
}

export const ConfigProvider: React.FC<ConfigProviderProps> = (props) => {
  const { children, themeName } = props;
  const lang = getLang();
  // eslint-disable-next-line no-nested-ternary
  const locale = lang === 'zh_CN' ? zhCN : lang === 'ru_RU' ? ruRU : enUS;
  const isDarkTheme = themeName === 'dark';

  return (
    <AntdConfigProvider
      locale={locale}
      theme={{
        algorithm: isDarkTheme ? theme.darkAlgorithm : theme.defaultAlgorithm,
      }}
    >
      {children}
    </AntdConfigProvider>
  );
};
