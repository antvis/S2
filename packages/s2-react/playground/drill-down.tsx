import { ORIGIN_FIELD, type PivotDataSet, type RawData } from '@antv/s2';
import type { PartDrillDownInfo } from '@antv/s2-shared';
import { forEach, random } from 'lodash';
import React from 'react';
import type { PartDrillDown } from '../src/components/sheets/interface';

const DrillDownFieldMap: Record<string, string[]> = {
  channel: ['物美', '华联'],
  sex: ['男', '女'],
};

export const partDrillDown: PartDrillDown = {
  drillConfig: {
    dataSet: [
      {
        name: '销售渠道',
        value: 'channel',
        type: 'text',
      },
      {
        name: '客户性别',
        value: 'sex',
        type: 'text',
      },
    ],
    extra: <div>test</div>,
  },
  // drillItemsNum: 1,
  fetchData: (meta, drillFields) =>
    new Promise<PartDrillDownInfo>((resolve) => {
      // 弹窗 -> 选择 -> 请求数据
      const dataSet = meta.spreadsheet.dataSet as PivotDataSet;
      const field = drillFields[0];

      const rowData = dataSet
        .getCellMultiData({
          query: meta.query!,
        })
        .filter(
          (item) =>
            item.getValueByField('type') && item.getValueByField('sub_type'),
        );

      const drillDownData: RawData[] = [];

      forEach(rowData, (data) => {
        const { number, sub_type: subType, type } = data[ORIGIN_FIELD];
        const number0 = random(50, number as number);
        const number1 = (number as number) - number0;
        const dataItem0: RawData = {
          ...meta.query,
          number: number0,
          sub_type: subType,
          type,
          [field]: DrillDownFieldMap[field][0],
        };

        drillDownData.push(dataItem0);
        const dataItem1: RawData = {
          ...meta.query,
          number: number1,
          sub_type: subType,
          type,
          [field]: DrillDownFieldMap[field][1],
        };

        drillDownData.push(dataItem1);
      });

      resolve({
        drillField: field,
        drillData: drillDownData,
      });
    }),
};
