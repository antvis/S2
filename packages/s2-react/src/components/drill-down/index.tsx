import React, { ReactNode, useEffect, useState } from 'react';
import { Button, ConfigProvider, Empty, Input, Menu } from 'antd';
import cx from 'classnames';
import { isEmpty } from 'lodash';
import {
  CalendarIcon,
  LocationIcon,
  SearchIcon,
  TextIcon,
} from '../icons/index';
import { i18n } from '@/common/i18n';

import './index.less';

export interface DataSet {
  icon?: React.ReactNode;
  name: string;
  value: string;
  type?: 'text' | 'location' | 'date';
  disabled?: boolean;
}

export interface DrillDownProps {
  className?: string;
  titleText?: string;
  searchText?: string;
  clearButtonText?: string;
  extra?: ReactNode;
  dataSet: DataSet[];
  drillFields?: string[];
  disabledFields?: string[];
  getDrillFields?: (drillFields: string[]) => void;
  setDrillFields?: (drillFields: string[]) => void;
}

export const DrillDown: React.FC<DrillDownProps> = ({
  className,
  titleText = i18n('选择下钻维度'),
  clearButtonText = i18n('恢复默认'),
  searchText = i18n('搜索字段'),
  extra,
  drillFields,
  dataSet,
  disabledFields,
  getDrillFields,
  setDrillFields,
  ...restProps
}) => {
  const PRE_CLASS = 's2-drill-down';
  const DRILL_DOWN_ICON_MAP = {
    text: <TextIcon />,
    location: <LocationIcon />,
    date: <CalendarIcon />,
  };

  const getOptions = () => {
    return dataSet.map((val: DataSet) => {
      const item = val;
      item.disabled = !!(disabledFields && disabledFields.includes(item.value));
      return item;
    });
  };

  const [options, setOptions] = useState<DataSet[]>(getOptions());

  const handleSearch = (e: any) => {
    const { value } = e.target;

    if (!value) {
      setOptions([...dataSet]);
    } else {
      const reg = new RegExp(value, 'gi');
      const result = dataSet.filter((item) => reg.test(item.name));
      setOptions([...result]);
    }
  };

  const handleSelect = (value: any) => {
    const key = value?.selectedKeys;
    if (getDrillFields) {
      getDrillFields([...key]);
    }
    if (setDrillFields) setDrillFields([...key]);
  };

  const handleClear = (e: React.MouseEvent<HTMLElement>) => {
    e.stopPropagation();
    if (getDrillFields) getDrillFields([]);
    if (setDrillFields) setDrillFields([]);
  };

  useEffect(() => {
    setOptions(getOptions());
  }, [disabledFields]);

  return (
    <ConfigProvider>
      <div className={cx(PRE_CLASS, className)} {...restProps}>
        <header className={`${PRE_CLASS}-header`}>
          <div>{titleText}</div>
          <Button
            type="link"
            disabled={isEmpty(drillFields)}
            onClick={handleClear}
          >
            {clearButtonText}
          </Button>
        </header>
        <Input
          className={`${PRE_CLASS}-search`}
          placeholder={searchText}
          onChange={handleSearch}
          onPressEnter={handleSearch}
          prefix={<SearchIcon />}
          allowClear
        />
        {isEmpty(options) && (
          <Empty
            imageStyle={{ height: '64px' }}
            className={`${PRE_CLASS}-empty`}
          />
        )}
        {extra}
        <Menu
          className={`${PRE_CLASS}-menu`}
          selectedKeys={drillFields}
          onSelect={handleSelect}
        >
          {options.map((option) => (
            <Menu.Item
              key={option.value}
              disabled={option.disabled}
              className={`${PRE_CLASS}-menu-item`}
              icon={
                option.icon ? option.icon : DRILL_DOWN_ICON_MAP[option.type]
              }
            >
              {option.name}
            </Menu.Item>
          ))}
        </Menu>
      </div>
    </ConfigProvider>
  );
};
