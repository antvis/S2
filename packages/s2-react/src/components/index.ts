export { BaseSheet } from './sheets/base-sheet';
export { TableSheet } from './sheets/table-sheet';
export { GridAnalysisSheet } from './sheets/grid-analysis-sheet';
export { StrategySheet } from './sheets/strategy-sheet';
export { AdvancedSort } from './advanced-sort';
export { DrillDown } from './drill-down';
export { TooltipComponent } from './tooltip';
export { Switcher } from './switcher';
export * from './sheets';
export * from './sheets/interface';
