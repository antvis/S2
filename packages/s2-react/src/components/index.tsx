import React from 'react';
import debounceRender from 'react-debounce-render';
import { BaseSheet } from './sheets/base-sheet';
import { TabularSheet } from './sheets/tabular-sheet';
import { TableSheet } from './sheets/table-sheet';
import { SpreadsheetProps } from './sheets/interface';
export { DrillDown } from './drill-down';
export { Switcher } from './switcher';
export { AdvancedSort } from './advanced-sort';
export * from './sheets';

export const SheetComponent: React.FC<SpreadsheetProps> = debounceRender(
  (props: SpreadsheetProps) => {
    const { sheetType } = props;
    switch (sheetType) {
      case 'table':
        return <TableSheet {...props} />;
      case 'tabular':
        return <TabularSheet {...props} />;
      default:
        return <BaseSheet {...props} />;
    }
  },
  100,
);
