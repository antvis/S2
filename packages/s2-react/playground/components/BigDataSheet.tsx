import { SpreadSheet, type S2DataConfig } from '@antv/s2';
import React from 'react';

import {
  SheetComponent,
  type SheetComponentOptions,
  type SheetComponentsProps,
} from '../../src';
import { usePlaygroundContext } from '../context/playground.context';

const s2Options: SheetComponentOptions = {
  width: 1600,
  height: 480,
  interaction: {
    scrollSpeedRatio: {
      vertical: 10,
      horizontal: 10,
    },
  },
};

export function generateRawData(
  row: Record<string, number>,
  col: Record<string, number>,
) {
  const res: Record<string, any>[] = [];
  const rowKeys = Object.keys(row);
  const colKeys = Object.keys(col);

  for (let i = 0; i < row[rowKeys[0]]; i++) {
    for (let j = 0; j < row[rowKeys[1]]; j++) {
      for (let m = 0; m < col[colKeys[0]]; m++) {
        for (let n = 0; n < col[colKeys[1]]; n++) {
          res.push({
            province: `p${i}`,
            city: `c${j}`,
            type: `type${m}`,
            subType: `subType${n}`,
            number: i * n,
          });
        }
      }
    }
  }

  return res;
}

const s2DataConfig: S2DataConfig = {
  fields: {
    rows: ['type', 'subType'],
    columns: ['province', 'city'],
    values: ['number'],
  },
  data: generateRawData(
    { province: 10, city: 100 },
    { type: 10, subType: 100 },
  ),
};

type CustomGridProps = Partial<SheetComponentsProps>;
const sheetType = 'pivot';

export const BigDataSheet = React.forwardRef<SpreadSheet, CustomGridProps>(
  (props, ref) => {
    const context = usePlaygroundContext();

    return (
      <SheetComponent
        {...props}
        {...context}
        sheetType={sheetType}
        dataCfg={s2DataConfig}
        options={s2Options}
        ref={ref}
      />
    );
  },
);
