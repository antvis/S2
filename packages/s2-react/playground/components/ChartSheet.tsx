/* eslint-disable import/no-extraneous-dependencies */
import { renderToMountedElement, stdlib } from '@antv/g2';
import { SpreadSheet, type ThemeCfg } from '@antv/s2';
import React from 'react';
import { ChartDataConfig } from '../../__tests__/data/data-g2-chart';
import {
  SheetComponent,
  type SheetComponentOptions,
  type SheetComponentsProps,
} from '../../src/components';
import { usePlaygroundContext } from '../context/playground.context';

const options: SheetComponentOptions = {
  height: 900,
  interaction: {
    copy: { enable: true, withFormat: true },
    brushSelection: {
      rowCell: true,
      colCell: true,
      dataCell: false,
    },
    selectedCellMove: true,
    selectedCellHighlight: true,
    selectedCellsSpotlight: true,
    hoverFocus: true,
    hoverHighlight: false,
    multiSelection: true,
    overscrollBehavior: 'none',
  },
  style: {},
};

const theme: ThemeCfg['theme'] = {
  dataCell: {},
};

const onDataCellRender: SheetComponentsProps['onDataCellRender'] = (cell) => {
  // 普通数值单元格正常展示
  if (!cell.isChartData()) {
    return;
  }

  const chartOptions = cell.getRenderChartOptions();

  // https://g2.antv.antgroup.com/manual/extra-topics/bundle#g2stdlib
  renderToMountedElement(chartOptions, {
    group: cell,
    library: stdlib(),
  });
};

export const ChartSheet: React.FC<
  Partial<SheetComponentsProps> & React.RefAttributes<SpreadSheet>
> = React.forwardRef((props, ref) => {
  const context = usePlaygroundContext();

  return (
    <SheetComponent
      {...props}
      {...context}
      sheetType="chart"
      themeCfg={{ theme }}
      dataCfg={ChartDataConfig}
      options={options}
      ref={ref}
      adaptive={true}
      onDataCellRender={onDataCellRender}
      header={{
        title: 'S2 & G2',
        description: '单元格内绘制 G2 图表',
        export: { open: true },
      }}
    />
  );
});
