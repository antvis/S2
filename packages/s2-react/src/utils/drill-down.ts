import { clone, filter, get, isEmpty, set } from 'lodash';
import {
  S2Options,
  HeaderActionIconProps,
  S2Event,
  SpreadSheet,
  Node,
  PivotDataSet,
  HeaderActionIcon,
  PartDrillDownDataCache,
  GEvent,
} from '@antv/s2';
import React from 'react';
import { PartDrillDownInfo, SheetComponentsProps } from '@/components';
import { i18n } from '@/common/i18n';

export interface DrillDownParams {
  // 行维度id
  rows: string[];
  // 下钻维度
  drillFields: string[];
  spreadsheet: SpreadSheet;
  // 下钻维值显示个数
  drillItemsNum?: number;
  // 下钻维度后获取数据
  fetchData?: (meta: Node, drillFields: string[]) => Promise<PartDrillDownInfo>;
}

export interface ActionIconParams {
  // 点击节点信息
  meta: Node;
  // 点击icon类型
  iconName: string;
  // 点击事件event
  event?: GEvent;
  spreadsheet: SpreadSheet;
  // 下钻维度的列表组件展示
  callback: (
    spreadsheet: SpreadSheet,
    cacheDrillFields: string[],
    disabledFields: string[],
    event?: GEvent,
  ) => void;
}

/**
 * 获取下钻缓存
 * @param spreadsheet
 * @param meta
 */
export const getDrillDownCache = (spreadsheet: SpreadSheet, meta: Node) => {
  const drillDownDataCache = spreadsheet.store.get(
    'drillDownDataCache',
    [],
  ) as PartDrillDownDataCache[];
  const cache = drillDownDataCache.find((dc) => dc.rowId === meta.id);
  return {
    drillDownDataCache: drillDownDataCache,
    drillDownCurrentCache: cache,
  };
};

/**
 * 点击下钻Icon的响应
 * @param params
 */
export const handleActionIconClick = (params: ActionIconParams) => {
  const { meta, spreadsheet, event, callback, iconName } = params;

  if (iconName === 'DrillDownIcon') {
    spreadsheet.store.set('drillDownNode', meta);
    const { drillDownDataCache, drillDownCurrentCache } = getDrillDownCache(
      spreadsheet,
      meta,
    );
    const cache = drillDownCurrentCache?.drillField
      ? [drillDownCurrentCache?.drillField]
      : [];
    const disabled = [];
    // 父节点已经下钻过的维度不应该再下钻
    drillDownDataCache.forEach((val) => {
      if (meta.id.includes(val.rowId) && meta.id !== val.rowId) {
        disabled.push(val.drillField);
      }
    });
    spreadsheet.emit(S2Event.GLOBAL_ACTION_ICON_CLICK, event);
    callback(spreadsheet, cache, disabled, event);
  }
};

export const handleDrillDownIcon = (
  props: SheetComponentsProps,
  spreadsheet: SpreadSheet,
  callback: (
    spreadsheet: SpreadSheet,
    cacheDownDrillFields: string[],
    disabledFields: string[],
    event?: GEvent,
  ) => void,
  drillDownIconRef: React.MutableRefObject<HeaderActionIcon>,
): S2Options => {
  const nextHeaderIcons =
    props.options.headerActionIcons?.filter(
      (icon) => icon !== drillDownIconRef.current,
    ) ?? [];

  if (props?.partDrillDown) {
    let displayCondition = props.partDrillDown?.displayCondition;
    if (isEmpty(displayCondition)) {
      let iconLevel = spreadsheet.store.get('drillDownActionIconLevel', -1);
      if (iconLevel < 0) {
        // 如果没有缓存，直接默认用叶子节点的层级
        iconLevel = get(props, 'dataCfg.fields.rows.length', 1) - 1;
        // 缓存配置之初的icon层级
        spreadsheet.store.set('drillDownActionIconLevel', iconLevel);
      }
      displayCondition = (meta: Node) => {
        // 只有数值置于列头且为树状分层结构时才支持下钻
        return (
          iconLevel <= meta.level &&
          spreadsheet.options.hierarchyType === 'tree' &&
          spreadsheet.dataCfg.fields.valueInCols &&
          meta.label !== i18n('总计')
        );
      };
    }

    const drillDownActionIcon = {
      belongsCell: 'rowCell',
      iconNames: ['DrillDownIcon'],
      defaultHide: true,
      displayCondition,
      action: (actionIconProps: HeaderActionIconProps) => {
        const { iconName, meta } = actionIconProps;
        if (iconName === 'DrillDownIcon') {
          spreadsheet.store.set('drillDownNode', meta);
          handleActionIconClick({
            ...actionIconProps,
            spreadsheet,
            callback,
          });
        }
      },
    };

    drillDownIconRef.current = drillDownActionIcon;
    nextHeaderIcons.push(drillDownActionIcon);
  } else if (!props?.partDrillDown && drillDownIconRef.current) {
    // clear previous icon ref
    drillDownIconRef.current = null;
  }

  set(props.options, 'headerActionIcons', nextHeaderIcons);

  return props.options;
};

export const handleDrillDown = (params: DrillDownParams) => {
  const { fetchData, spreadsheet, drillFields, drillItemsNum } = params;
  spreadsheet.store.set('drillItemsNum', drillItemsNum);
  const meta = spreadsheet.store.get('drillDownNode');
  const { drillDownDataCache, drillDownCurrentCache } = getDrillDownCache(
    spreadsheet,
    meta,
  );
  let newDrillDownDataCache = clone(drillDownDataCache);
  // 如果当前节点已有下钻缓存，需要清除
  if (drillDownCurrentCache) {
    newDrillDownDataCache = filter(
      drillDownDataCache,
      (cache) => cache.rowId !== meta.id,
    );
  }
  fetchData(meta, drillFields).then((info) => {
    const { drillData, drillField } = info;
    (spreadsheet.dataSet as PivotDataSet).transformDrillDownData(
      drillField,
      drillData,
      meta,
    );

    if (!isEmpty(drillData)) {
      // 缓存到表实例中
      const drillLevel = meta.level + 1;
      const newDrillDownData = {
        rowId: meta.id,
        drillLevel,
        drillData,
        drillField,
      };
      newDrillDownDataCache.push(newDrillDownData);
      spreadsheet.store.set('drillDownDataCache', newDrillDownDataCache);
    }

    // 重置当前交互
    spreadsheet.interaction.reset();
    spreadsheet.render(false);
  });
};
