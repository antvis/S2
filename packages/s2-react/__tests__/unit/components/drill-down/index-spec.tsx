import React from 'react';
import { render, screen } from '@testing-library/react';
import { DrillDown } from '@/components/drill-down';

describe('DrillDown Component Tests', () => {
  test('should render component', () => {
    const { asFragment, container } = render(
      <DrillDown className="test" extra="extra" dataSet={[]} />,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('.test')).toBeDefined();
    expect(screen.getByText('extra')).toBeDefined();
  });
});
