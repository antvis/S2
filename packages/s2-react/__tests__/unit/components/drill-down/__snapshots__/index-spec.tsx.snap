// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrillDown Component Tests should render component 1`] = `
<DocumentFragment>
  <div
    class="antv-s2-drill-down test"
  >
    <header
      class="antv-s2-drill-down-header"
    >
      <div>
        选择下钻维度
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-p7e5j5 ant-btn-link"
        disabled=""
        type="button"
      >
        <span>
          恢复默认
        </span>
      </button>
    </header>
    <span
      class="ant-input-affix-wrapper antv-s2-drill-down-search css-dev-only-do-not-override-p7e5j5 css-dev-only-do-not-override-p7e5j5"
    >
      <span
        class="ant-input-prefix"
      >
        <div />
      </span>
      <input
        class="ant-input css-dev-only-do-not-override-p7e5j5"
        placeholder="搜索字段"
        type="text"
        value=""
      />
      <span
        class="ant-input-suffix"
      >
        <span
          class="ant-input-clear-icon ant-input-clear-icon-hidden"
          role="button"
          tabindex="-1"
        >
          <span
            aria-label="close-circle"
            class="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              fill-rule="evenodd"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
    <div
      class="antv-s2-drill-down-extra"
    >
      extra
    </div>
    <div
      class="css-dev-only-do-not-override-p7e5j5 ant-empty antv-s2-drill-down-empty"
    >
      <div
        class="ant-empty-image"
        style="height: 64px;"
      >
        <svg
          height="152"
          viewBox="0 0 184 152"
          width="184"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              transform="translate(24 31.67)"
            >
              <ellipse
                cx="67.797"
                cy="106.89"
                fill="#F5F5F7"
                fill-opacity=".8"
                rx="67.797"
                ry="12.668"
              />
              <path
                d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
                fill="#AEB8C2"
              />
              <path
                d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                fill="url(#linearGradient-1)"
                transform="translate(13.56)"
              />
              <path
                d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
                fill="#F5F5F7"
              />
              <path
                d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
                fill="#DCE0E6"
              />
            </g>
            <path
              d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
              fill="#DCE0E6"
            />
            <g
              fill="#FFF"
              transform="translate(149.65 15.383)"
            >
              <ellipse
                cx="20.654"
                cy="3.167"
                rx="2.849"
                ry="2.815"
              />
              <path
                d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
              />
            </g>
          </g>
        </svg>
      </div>
      <div
        class="ant-empty-description"
      >
        No data
      </div>
    </div>
    <ul
      class="ant-menu ant-menu-root ant-menu-vertical ant-menu-light antv-s2-drill-down-menu css-dev-only-do-not-override-p7e5j5"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    />
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>
</DocumentFragment>
`;
